### DISCUSSION TITLE
[📚] Required Firebase Web JS Version

### DISCUSSION THREAD

**Discussion by rayliverified** [firebase/flutterfire#5786] [General]
The Firebase documentation and FlutterFire documentation conflict on the version of Firebase to use on the web. 

Firebase documentation (https://firebase.google.com/docs/web/setup#from-the-cdn):

```

  <!-- Firebase App (the core Firebase SDK) is always required and must be listed first -->
  <script src="https://www.gstatic.com/firebasejs/8.1.2/firebase-app.js"></script>

  <!-- If you enabled Analytics in your project, add the Firebase SDK for Analytics -->
  <script src="https://www.gstatic.com/firebasejs/8.1.2/firebase-analytics.js"></script>

  <!-- Add Firebase products that you want to use -->
  <script src="https://www.gstatic.com/firebasejs/8.1.2/firebase-auth.js"></script>
  <script src="https://www.gstatic.com/firebasejs/8.1.2/firebase-firestore.js"></script>

```

FlutterFire documentation (https://firebase.flutter.dev/docs/installation/web):

```
    <script src="https://www.gstatic.com/firebasejs/7.20.0/firebase-app.js"></script>
```

Please add documentation and guidance around the supported version and planned upgrades. 

There is a challenge in that the Firebase web dependencies are currently manually added to the index.html file so they are not automatically kept in sync with the FlutterFire SDK version. 

Without proper guidance, this could become a big issue down the road.

**Comment by rayliverified** [comment]
Without specifying a version results problems like the following at
https://github.com/FirebaseExtended/flutterfire/issues/4127
Each FlutterFire release should be locked to a Firebase JS SDK version.

**Comment by rayliverified** [comment]
Thank you for updating to the latest Firebase JS version.
However, how will they be kept in sync? 
Is that still currently a manual process? If so, then this issue should remain open.

**Comment by Salakar** [comment]
> However, how will they be kept in sync?

This is still an open discussion. We'd like to manage the versions of the underlying JS SDK like we do already on Android/iOS/macOS - but as far as we're aware there isn't yet a good way of injecting these scripts in via Dart so we can internally control versions. Open to suggestions
